src/StreamJsonRpc.Tests/JsonRpcWithFatalExceptionsTests.cs

```diff
@@ -219,7 +204,7 @@ public Task AsyncMethodThatThrowsBeforeYield(string message)
         {
             await Task.Run(() => throw new Exception(message));
 
-            return "never will return";
+            return await Task.FromResult("never will return");
```

Comment:
I'm curious why you're using `return await Task.FromResult(x)` instead of just `return x`. I can't think of any difference that makes at runtime and it seems more verbose.